gflownet/utils/batch.py

```diff
@@ -0,0 +1,288 @@
+                    """
+                    self.parents.append(parents)
+                    self.parents_actions.append(parents_a)
+                if self.loss == "trajectorybalance":
+                    self.masks_invalid_actions_backward.append(
+                        env.get_mask_invalid_actions_backward(
+                            env.state, env.done, [action]
+                        )
+                    )
+            else:
+                if env.done:
+                    self.states.append(env.state)
+                    self.env_ids.append(env.id)
+                    self.steps.append(env.n_actions)
+
+    def process_batch(self):
+        """
+        Process internal lists to more convenient formats:
+        - converts and stacks list to a single torch tensor
+        - computes trajectory indicies (indecies of the states in self.states corresponding to each trajectory)
+        - if needed, computes states and parents in policy formats (stored in self.states_policy, self.parents_policy)
+        """
+        self._process_states()
+        self.env_ids = tlong(self.env_ids, device=self.device)
+        self.steps = tlong(self.steps, device=self.device)
+        self._process_trajectory_indices()
+        # process other variables, if we are in the train mode and recorded them
+        if len(self.actions) > 0:
+            self.actions = tfloat(
+                self.actions, device=self.device, float_type=self.float
+            )
+            self.done = tbool(self.done, device=self.device)
+            self.masks_invalid_actions_forward = tbool(
+                self.masks_invalid_actions_forward, device=self.device
+            )
+            if self.loss == "flowmatch":
+                self.parents_state_idx = tlong(
+                    sum([[idx] * len(p) for idx, p in enumerate(self.parents)], []),
+                    device=self.device,
+                )
+                self.parents_actions = torch.cat(
+                    [
+                        tfloat(x, device=self.device, float_type=self.float)
+                        for x in self.parents_actions
+                    ]
+                )
+            elif self.loss == "trajectorybalance":
+                self.masks_invalid_actions_backward = tbool(
+                    self.masks_invalid_actions_backward, device=self.device
+                )
+            self._process_parents()
+        self.is_processed = True
+
+    def _process_states(self):
+        """
+        Converts self.statees from list to torch tensor, computes states in the policy format
+        (stored in self.states_policy)
+        """
+        self.states_policy = self.states2policy()
+        self.states = tfloat(self.states, device=self.device, float_type=self.float)
+
+    def states2policy(self, states=None, env_ids=None):
+        """
+        Converts states from a list of states in gflownet fromat to a tensor of states in policy format
```

Comment:
format

Reply:
could you elaborate?

Reply:
My bad, I meant that "fromat" should be changed to "format".

Reply:
My bad, I meant that "fromat" should be changed to "format".

Reply:
fixed :) thanks for pointing out, I'm having a lot of typos in my English 😅